### ISSUE TITLE
Add/edit rows

### ISSUE THREAD

**Issue by Rid** [mtxr/vscode-sqltools#298]
**Describe the solution you'd like**
At the moment we have to use a 3rd party tool for manipulating existing tables, such as for quick table row edits or additions. It would be great if we had one tool inside VS code for doing this.

I have been using `@devexpress/dx-react-grid-material-ui` myself in another project in uncontrolled mode, it's quite simple to add this kind of manipulation. I've not had a great chance to look at this project's code to ascertain what the dependencies would be regarding the different dialects, have you thought about this?

**Describe alternatives you've considered**
Previously we've been using DataGrip from jetbrains:
https://www.jetbrains.com/datagrip/features/data_editor.html

It has many features such as edit/addition/clone.

**Comment by mtxr** [comment]
This is upcoming!

I've started a new PR to switch from react-table to @devexpress/dx-react-grid-material-ui as seen on #297 .

I'm now working on add the vscode color scheme to it and I'll release it. 

Add and edit rows will be done right after that, because it requires some work on the dialects to respect the changes required by the user. SQLTools needs to be very careful with this kind of change on the DB. Queries are written by the user, we just execute it. But to add and edit rows from the UI needs to be 100% reliable on all dialects

**Comment by mtxr** [comment]
Just moved to react-grid. Next step is to allow editing :tada: :tada: :tada:

**Comment by Rid** [comment]
Great job @mtxr!

**Comment by regniblod** [comment]
Hello! Any updates on this? Is one of the few things left keeping me from moving from IntelliJ IDEs to VSCode.

**Comment by Gruntfuggly** [comment]
Just found this extension and it seemed awesome, until I needed to modify a value in a table. Was hoping to ditch Squirrel and use this...

**Comment by mtxr** [comment]
@Gruntfuggly as awesome as todo-tree? 🤣 I use yours as well

**Comment by Gruntfuggly** [comment]
Lol. Glad it's helping somebody! I think I'm still going to be able to get rid of squirrel, but i assume the only way to update the table at the moment is via a query?

**Comment by mtxr** [comment]
Yeah, for now. 

I had to do some changes in the extension and the way we create queries automatically in the extension to make it possible. So I expect this to be the next major feature coming.

**Comment by fela93** [comment]
Great job @mtxr!
Any update when this feature will be available?, very required feature

Thanks

**Comment by dcrystalj** [comment]
@mtxr  any code available already. we can contribute as this is kind of must have feature

**Comment by smaartdata** [comment]
Any update?

**Comment by Phara0h** [comment]
Would love to have this feature.

**Comment by bfine9618** [comment]
I would use this extension exclusively with this feature

**Comment by emil14** [comment]
Any updates? This one is extremely needed!

**Comment by miztizm** [comment]
Hello, any news on this feature ?

**Comment by Enubia** [comment]
I'd also like to see this one added, would be a huge benefit!

**Comment by dcrystalj** [comment]
I ended using this plugin https://marketplace.visualstudio.com/items?itemName=cweijan.vscode-database-client2

**Comment by arifur** [comment]
Any update on this?

**Comment by gjsjohnmurray** [comment]
I'm not aware of any in-progress work in this area, nor of plans to start.

**Comment by bombillazo** [comment]
It seems like this is the essential feature missing for this extension.

**Comment by Tyrenn** [comment]
Would love to have this feature

**Comment by piavgh** [comment]
@mtxr : hi, is there any ongoing progress with this issue?

**Comment by gjsjohnmurray** [comment]
@piavgh none that I am aware of.

**Comment by Ioann44** [comment]
No updates still?

**Comment by gjsjohnmurray** [comment]
@Ioann44 no. If you are in a position to fund development of this feature please get in touch.

**Comment by tommyhetterscheid17** [comment]
I would like this as well

**Comment by patrickwolf** [comment]
Unfortunate that this is missing. Would really be the one stop extension for DB with editing  included

**Comment by Blackspell01** [comment]
Is this extension still maintained?

**Comment by gjsjohnmurray** [comment]
> Is this extension still maintained?

Yes

**Comment by Blackspell01** [comment]
Im using the database client extension atm but it feels bloated, is editing tables planned in the near future? Because it seems to be an open feature request for 7 years now...

**Comment by gjsjohnmurray** [comment]
As maintainers we're not currently planning to implement this requested enhancement, nor are we aware of any incoming community contributions to do that.

**Comment by Blackspell01** [comment]
Hmm, that's pretty sad
